src/index.ts

```diff
@@ -131,6 +133,11 @@ async function init(): Promise<void> {
     monitorCoreRpcConnection().catch(error => {
       logger.error(`Error monitoring RPC connection: ${error}`, error);
     });
+
+    // TODO: register an exit-handler for the tokenProcessorQueue instance
+    const tokenMetadataProcessor = new TokensProcessorQueue(db, configuredChainID);
```

Comment:
You could use the newly added shutdown config handlers here for this TODO

Reply:
Good idea, done!